src/libexpr/eval.cc

```diff
@@ -1575,36 +1573,36 @@ void ExprConcatStrings::eval(EvalState & state, Env & env, Value & v)
            since paths are copied when they are used in a derivation),
            and none of the strings are allowed to have contexts. */
         if (first) {
-            firstType = vTmp.type;
+            firstType = vTmp.normalType();
```

Comment:
The `ExprConcatStrings::eval` function is an interesting place where the new normal form type has to be used to ensure it keeps working the same way